lib/Validator.ts

```diff
@@ -44,24 +77,56 @@ export default class Validator {
    * @returns list of validated events
    */
   validateEventOrder(eventsList: any[]): any[] {
-    if(!eventsList) return eventsList;
+    if (!eventsList) return eventsList;
     this.logger.debug(`Number of events: ${eventsList.length}`);
     let events: any[] = [];
     try {
-      eventsList[0]['valid'] = true;
+      if (eventsList[0]['event'] !== 'init') {
+        eventsList[0]['valid'] = false;

```

Comment:
The code is currently mutating the incoming array, which is not good, since it can unpredictably affect unrelated code making use of this package, if it holds references to the original eventsList array.

Non mutating code could (for example) be achieved by mapping the incoming array to a new array of validated events, instead of operating directly on the incoming array: `const validatedEvents = eventsList.map((event: TPlayerAnalyticsEvent) => ValidatedEvent)`.